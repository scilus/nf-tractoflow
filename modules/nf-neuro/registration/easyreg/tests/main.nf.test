nextflow_process {

    name "Test Process REGISTRATION_EASYREG"
    script "../main.nf"
    process "REGISTRATION_EASYREG"

    tag "modules"
    tag "modules_nfcore"
    tag "registration"
    tag "registration/easyreg"

    tag "subworkflows"
    tag "subworkflows/load_test_data"


    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "freesurfer.zip", "registration.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("registration - easyreg") {
        config "./nextflow.config"
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        t1: it.simpleName == "freesurfer"
                        b0: it.simpleName == "registration"
                    }
                ch_t1 = ch_split_test_data.t1.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/t1.nii.gz")
                    ]
                }
                ch_b0 = ch_split_test_data.b0.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/b0.nii.gz"),
                        [],
                        []
                    ]
                }
                input[0] = ch_t1
                    .join(ch_b0)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.ref_seg.get(0).get(1)).name,
                    file(process.out.flo_seg.get(0).get(1)).name,
                    niftiMD5SUM(process.out.ref_reg.get(0).get(1), 6),
                    niftiMD5SUM(process.out.flo_reg.get(0).get(1), 6),
                    niftiMD5SUM(process.out.fwd_field.get(0).get(1), 6),
                    niftiMD5SUM(process.out.bak_field.get(0).get(1), 6),
                    process.out.versions
                ).match() }
            )
        }
    }
}
