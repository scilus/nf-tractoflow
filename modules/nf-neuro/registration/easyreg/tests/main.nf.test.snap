{
    "registration - easyreg": {
        "content": [
            "test_reference_segmentation.nii.gz",
            "test_floating_segmentation.nii.gz",
            "test_reference_registered.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,1501221fe23cd62bfdafb33367cadf4d",
            "test_floating_registered.nii.gz:md5:header,ec5893cd9ea024e630c4444bd914c331,data,d75ae3fc935cd5cc70ea6797a4c70775",
            "test_forward_field.nii.gz:md5:header,0db0a80786ff39864cc17506ed1a0146,data,e4fa6c626729cbf236c34680e5c76df4",
            "test_backward_field.nii.gz:md5:header,74c92ee4cd3c4abfecf3da6cdb1d4650,data,8f26afa5c3de21469466213498542486",
            [
                "versions.yml:md5,4661210880c42a986923e8257f64c760"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T16:01:24.00559"
    }
}